ui/src/actions/login.js

```diff
@@ -158,6 +162,20 @@ export function getInitialState() {
     ]);
 
     dispatch(setInitialState(Object.assign({}, ...initialStateSlices)));
+
+    const chatMessagesData = initialStateSlices.find(
+      (slice) => slice.chatMessages,
+    );
+    if (chatMessagesData?.chatMessages) {
+      const { user } = getState().login;
+      dispatch(
+        processFetchedChatMessages(
+          chatMessagesData.chatMessages.messages,
+          user.username,
+        ),
+      );
+    }
```

Comment:
There's already a `setInitialState` action being dispatched with all the data, so it would be simpler I think to process the initial chat messages with `processChatMessageRecord` in the `SET_INITIAL_STATE` case block of the `remoteAccess` reducer.